rotkehlchen/api/rest.py

```diff
@@ -4546,6 +4546,7 @@ def _detect_ethereum_tokens(
         evmtokens = EvmTokens(
             database=self.rotkehlchen.data.db,
             manager=self.rotkehlchen.chain_manager.ethereum,
+            makerdao_vaults_module=self.rotkehlchen.chains_aggregator.get_module('makerdao_vaults'),  # noqa: E501
```

Comment:
This looks wrong, for example if you have `aave` module active but not the `maker dao` module this will return `None` and you won't fetch the information for the ds proxies that have aave positions. Or am I misunderstanding something?

Checking the code maybe the best approach is to split the `HasDSProxy` interface and move the logic that only queries the chain to a different class?

Reply:
Hmmmm I used the same I approach that I saw for querying zerion https://github.com/rotki/rotki/blob/e44c8654fb3e3bd4d8b890f0d03a1b952af87229/rotkehlchen/chain/manager.py#L1231. I guess makerdao module is almost always on and because of that it's fine. But it's just a guess.